examples/svg_viewer/src/main.rs

```diff
@@ -0,0 +1,334 @@
+type DepthFormat = gfx::format::DepthStencil;
+
+gfx_defines!{
+    vertex GpuFillVertex {
+        position: [f32; 2] = "a_position",
+        color: [f32; 4] = "a_color",
+    }
+
+    constant Globals {
+        zoom: [f32; 2] = "u_zoom",
+        pan: [f32; 2] = "u_pan",
+    }
+
+    pipeline fill_pipeline {
+        vbo: gfx::VertexBuffer<GpuFillVertex> = (),
+        out_color: gfx::RenderTarget<ColorFormat> = "out_color",
+        constants: gfx::ConstantBuffer<Globals> = "Globals",
+    }
+}
+
+fn gpu_color(color: Color, opacity: f32) -> [f32; 4] {
+    [
+        f32::from(color.red) / 255.0,
+        f32::from(color.green) / 255.0,
+        f32::from(color.blue) / 255.0,
+        opacity
+    ]
+}
+
+// A very simple vertex constructor that only outputs the vertex position
+struct VertexCtor {
+    fill: Color
+}
+impl VertexConstructor<tessellation::FillVertex, GpuFillVertex> for VertexCtor {
+    fn new_vertex(&mut self, vertex: tessellation::FillVertex) -> GpuFillVertex {
+        assert!(!vertex.position.x.is_nan());
+        assert!(!vertex.position.y.is_nan());
+        GpuFillVertex {
+            // (ugly hack) tweak the vertext position so that the logo fits roughly
+            // within the (-1.0, 1.0) range.
+            position: (vertex.position * 0.0145 - vec2(1.0, 1.0)).to_array(),
+            color: gpu_color(self.fill, 1.0)
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug)]
+struct Scene {
+    zoom: f32,
+    pan: [f32; 2]
+}
+
+// extract the relevant globals from the scene struct
+impl From<Scene> for Globals {
+    fn from(scene: Scene) -> Self {
+        Globals {
+            zoom: [scene.zoom, scene.zoom],
+            pan: scene.pan
+        }
+    }
+}
+
+struct SvgPath {
+    d: String,
+    fill: Color
+}
+
+impl Default for SvgPath {
+    fn default() -> Self {
+        SvgPath {
+            d: String::new(),
+            fill: Color::new(0,0,0)
+        }
+    }
+}
+
+fn main() {
+    // Parse CLI arguments
+    let cli_args = App::new("Lyon CLI Renderer")
+                          .arg(Arg::with_name("input")
+                               .index(1)
+                               .value_name("FILE")
+                               .help("SVG file to render")
+                               .takes_value(true))
+                          .get_matches();
+
+    let filepath = cli_args.value_of("input").expect("Missing input file");
+
+    // read the SVG file into a string
+    let mut input_buffer = String::new();
+    let mut file = File::open(filepath)
+        .expect("Error opening input file!");
+    file.read_to_string(&mut input_buffer)
+        .expect("Error reading input file!");
+
+    // process the SVG tokens and generate an array of paths
+    let mut last_tag = None;
+    let mut svg_paths = Vec::new();
+    let mut current_path = SvgPath::default();
+    let mut tokens = Tokenizer::from_str(&input_buffer).tokens();
+    for token in &mut tokens {
+        match token {
```

Comment:
A lot of SVG drawing software like inkscape tend to apply transforms to most elements instead of specifying paths in the reference coordinate system of the document. It would be great to support parsing transform attributes here and either apply the transform in the VertexConstructor, or in the vertex shader.

Reply:
Thanks for pointing that out.  I'm not super familiar with SVG transforms but I'll look into it.